File: June 2022/June_29.java
class Solution {
    public int[][] reconstructQueue(int[][] people) {
        int n = people.length;
        Arrays.sort(people, (a, b) -> a[1] != b[1] ? a[1] - b[1] : a[0] - b[0]);
        List<int[]> list = new ArrayList<>();
        int i = 0;
        while(i < n && people[i][1] == 0)
            list.add(people[i++]);
        while(i < n){
            int count = 0;
            for(int j = 0; j < i; j++){
                if(people[i][0] <= list.get(j)[0])
                    count++;
                if(count == people[i][1] + 1){
                    list.add(j, people[i]);
                    break;
                }
            }
            if(count <= people[i][1])
                list.add(people[i]);
            i++;
        }
        
        int[][] ans = new int[n][2];
        for(i = 0; i < n; i++)
            ans[i] = list.get(i);
        return ans;
    }
}

File: June 2022/June_29.cpp
class Solution {
public:
    static bool cmp(vector<int> &a, vector<int> &b){
        if(a[1] != b[1])
            return a[1] < b[1];
        return a[0] < b[0];
    }
    
    vector<vector<int>> reconstructQueue(vector<vector<int>>& people) {
        int n = people.size();
        sort(people.begin(), people.end(), cmp);
        vector<vector<int>> ans;
        int i = 0;
        while(i < n && people[i][1] == 0)
            ans.push_back(people[i++]);
        while(i < n){
            int count = 0;
            for(int j = 0; j < i; j++){
                if(people[i][0] <= ans[j][0])
                    count++;
                if(people[i][1] + 1 == count){
                    ans.insert(ans.begin() + j, people[i]);
                    break;
                }
            }
            if(people[i][1] >= count)
                ans.push_back(people[i]);
            i++;
        }
        return ans;
    }
};
